Description: Retain exit code in cloud-init status for recoverable errors
 (LP: #2048522).
Author: Alberto Contreras <alberto.contreras@canonical.com>
Last-Update: 2024-01-08
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/cloudinit/cmd/status.py
+++ b/cloudinit/cmd/status.py
@@ -255,7 +255,7 @@ def handle_status_args(name, args) -> in
         return 1
     # Recoverable error
     elif details.condition_status == ConditionStatus.DEGRADED:
-        return 2
+        return 0
     return 0
 
 
--- a/tests/unittests/cmd/test_status.py
+++ b/tests/unittests/cmd/test_status.py
@@ -730,7 +730,7 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr
                 },
                 None,
                 MyArgs(long=False, wait=False, format="json"),
-                2,
+                0,
                 {
                     "_schema_version": "1",
                     "boot_status_code": "enabled-by-kernel-command-line",
